diag_arp_flux/diag_arp_flux.py

```diff
@@ -0,0 +1,351 @@
+    for addr in ip_in_int:
+        if addr < 0 or addr > 255:
+            raise Exception("Invalid IPv4 address: {}".format(str_ip_address))
+    return ip_in_int
+
+
+def scan_all_interfaces():
+    """
+    Scan all addresses on all interfaces, then fill dict_subnet
+    with subnet information
+    """
+    global dict_subnet
+    global TABLES_TO_DEFINE
+    TABLES_TO_DEFINE = []
+    print "Scanning all interfaces..."
+    ifs = netifaces.interfaces()
+    for interface in ifs:
+        obj_if = Interface(interface)
+
+        if_name = obj_if.name
+        for dict_ipv4 in obj_if.ifv4:
+            try:
+                if_addr = dict_ipv4["addr"]
+                if_mask = dict_ipv4["netmask"]
+            except KeyError, e:
+                continue
+            subnet = get_subnet_in_str(if_addr, if_mask)
+            print "    {} ip:{} netmask:{} in subnet:{}".format(if_name, if_addr, if_mask, subnet)
+            if subnet not in dict_subnet:
+                dict_subnet[subnet] = []
+            dict_subnet[subnet].append({"if_name": if_name, "ip": if_addr})
+
+    print "Subnet statistics:"
+    print json.dumps(dict_subnet, indent=4, sort_keys=True)
+
+    subnet_ct = 0
+    print "Subnets has arp flux risks:"
+    for subnet in dict_subnet:
+        if len(dict_subnet[subnet]) >= 2:
+            subnet_ct += 1
+            print "    {}".format(subnet)
+
+            for info in dict_subnet[subnet]:
+                table_name = "{}{}".format(TABLE_PREFIX, info["if_name"])
+                if table_name not in TABLES_TO_DEFINE:
+                    TABLES_TO_DEFINE.append(table_name)
+    if subnet_ct == 0:
+        print "    None"
+
+
+def route_table_add_interface_rule():
+    print "Add IP route rules for table {} ...".format(TABLE_PREFIX)
+    for subnet in dict_subnet:
+        if len(dict_subnet[subnet]) >= 2:
+            # add rules
+            # 1. ip rule add from <ip> table <table>
+            # 2. ip route add <subnet> via <ip> dev <interface> table <table>
+            for info in dict_subnet[subnet]:
+                cmd_ip_rule = "ip rule add from {} table {}".format(info["ip"], TABLE_PREFIX+info["if_name"])
+                ret, rsp = run_command(cmd_ip_rule)
+                if ret != 0:
+                    print "    Fail to run: {}".format(cmd_ip_rule)
+                    exit(1)
+                print "    {}".format(cmd_ip_rule)
+
+                cmd_ip_route_on_nic = "ip route add {} via {} dev {} table {}".\
+                    format(subnet, info["ip"], info["if_name"], TABLE_PREFIX+info["if_name"])
+                ret, rsp = run_command(cmd_ip_route_on_nic)
+                if ret != 0:
+                    print "    Fail to run: {}".format(cmd_ip_route_on_nic)
+                    exit(1)
+                print "    {}".format(cmd_ip_route_on_nic)
+
+                gateway = default_gateway(subnet)
+                cmd_ip_route_default = "ip route add default via {} dev {} table {}".\
```

Comment:
[minor]run one time is enough for one interface. it could be move to upper layer iteration.